engine/src/main/java/net/jqwik/engine/properties/FeatureExtractor.java

```diff
@@ -20,12 +20,17 @@ default Object applySafe(T t) {
 	}
 
 	default boolean isUniqueIn(T value, Collection<T> elements) {
-		Set<Object> elementFeatures = elements.stream().map(this::applySafe).collect(Collectors.toSet());
-		return !elementFeatures.contains(this.applySafe(value));
+		if (this == identity()) {
+			return !elements.contains(value);
+		}
+		Object feature = applySafe(value);
+		return elements.stream()
+					   .map(this::applySafe)
+					   .noneMatch(x -> Objects.equals(x, feature));
```

Comment:
Is this `applySafe` still important?

I am inclined `applySafe` should be replaced with just `apply`.

Reply:
Don't think so. Collections are allowed to have null values.

Look at this test in `ListArbitraryTests`.

	@Example
	void uniqueElementsWithNull(@ForAll Random random) {
		ListArbitrary<Integer> listArbitrary =
			Arbitraries.integers().between(1, 1000).injectNull(0.5)
					   .list().ofMaxSize(20)
					   .uniqueElements(i -> i % 100);

		RandomGenerator<List<Integer>> generator = listArbitrary.generator(1000, true);

		assertAllGenerated(generator, random, list -> {
			assertThat(isUniqueModulo(list, 100)).isTrue();
		});
	}

Reply:
Well, if `applySafe` is not needed, then I am inclined to remove it.

Reply:
But it is needed. Otherwise, the test will fail.

Reply:
I thought I did try removing `applySafe` and it revealed no test failures, so I asked why that was needed. Let me try again.